searchlib/src/vespa/searchlib/queryeval/exact_nearest_neighbor_iterator.cpp

```diff
@@ -19,7 +19,7 @@ namespace search::queryeval {
  * Keeps a heap of the K best hit distances.
  * Currently always does brute-force scanning, which is very expensive.
  **/
-template <bool strict, bool has_filter, bool has_single_subspace>
+template <bool strict, bool has_filter, bool readonly_distance_heap, bool has_single_subspace>
```

Comment:
No need to readonly_distance_heap as a template argument. It is only used in unpack, which a very cold method.

Reply:
Perfect